restli-server/src/test/java/com/linkedin/restli/internal/server/TestRestLiRouter.java

```diff
@@ -0,0 +1,237 @@
+
+    Assert.assertEquals(e.getStatus(), HttpStatus.S_404_NOT_FOUND.getCode());
+  }
+
+  @Test
+  public void failsOnVeryShortUriPath() throws URISyntaxException
+  {
+
+    final TestSetup setup = new TestSetup();
+    final RestLiRouter router = setup._router;
+    final ServerResourceContext context = setup._context;
+
+    doReturn(new URI("/")).when(context).getRequestURI();
+
+    final RoutingException e = runAndCatch(() -> router.process(context), RoutingException.class);
+
+    Assert.assertEquals(e.getStatus(), HttpStatus.S_404_NOT_FOUND.getCode());
+  }
+
+  // ----------------------------------------------------------------------
+  // helper members
+  // ----------------------------------------------------------------------
+
+  @SuppressWarnings("unchecked")
+  private static <X extends Throwable> X runAndCatch(ThrowingRunnable<X> runnable, Class<X> clazz) {
+    try {
+      runnable.run();
+    } catch (Throwable t) {
+      if (clazz.isAssignableFrom(t.getClass())) {
+        return (X) t;
+      }
+      throw new IllegalStateException(
+          String.format("Expected an exception of type '%s' but caught a different one: %s",
+              clazz.getSimpleName(), t.getMessage()), t);
+    }
+    throw new IllegalStateException(
+        String.format("Expected an exception of type '%s' but none was caught", clazz.getSimpleName()));
+  }
+
+  @RestLiCollection(name = "root", keyName = "rootId")
+  private static class RootResource extends CollectionResourceTemplate<Long, EmptyRecord>
+  {
+    @RestMethod.Get
+    public EmptyRecord get(@PathKeyParam("rootId") Long id)
+    {
+      return new EmptyRecord();
+    }
+  }
+
+  private static final class TestSetup
+  {
+    private final String _childPath;
+    private final RestLiConfig _config;
+    private final ServerResourceContext _context;
+    private final Class<?> _keyClass;
+    private final String _keyName;
+    private final DataMap _parameters;
+    private final MutablePathKeys _pathKeys;
+    private final Map<String, ResourceModel> _pathToModelMap;
+    private final String _resourceName;
+    private final RequestContext _requestContext;
+    private final ResourceModel _rootModel;
+    private final String _rootPath;
+    private final RestLiRouter _router;
+
+    private TestSetup() {
+      _keyName = "rootId";
+      _keyClass = Long.class;
+      _resourceName = "root";
+      _childPath = "/child";
+      _rootPath = "/root";
+      _parameters = new DataMap();
+      _context = mock(ServerResourceContext.class);
+      _pathKeys = mock(MutablePathKeys.class);
+      _rootModel = RestLiAnnotationReader.processResource(RootResource.class);
+      _pathToModelMap = new HashMap<>();
+      _config = new RestLiConfig();
+      _router = new RestLiRouter(_pathToModelMap, _config);
+      _requestContext = new RequestContext();
+    }
+
+    private void mockCommon() {
```

Comment:
Can this be setup using "beforeMethod"?

Reply:
Yes it can, along with the test setup creation. Will make the change.

Reply:
Actually, I take it back: `TestRestLiRouter#failsOnRootResourceNotFound` requires no mocking.